src/libecalc/common/utils/rates.py

```diff
@@ -156,6 +156,23 @@ def convert_none_to_nan(cls, v: float, field: ModelField) -> TimeSeriesValue:
             return math.nan
         return v
 
+    @validator("values", pre=True)
+    def timesteps_values_one_to_one(cls, v: list, values, field: ModelField):
+        nr_timesteps = len(values["timesteps"])
+        nr_values = len(v)
+
+        if not cls.__name__ == TimeSeriesVolumes.__name__:
```

Comment:
Can you add an additional test for TimeSeriesVolumes, that the `len(values) == len(timesteps) -1` ?

I think it was where i saw a lot of errors when working with regularity. I dont think i fixed anything

Reply:
TimeSeriesVolumes: initially len(values) == len(timesteps) - 1. 

However, after TimeSeriesVolumes.reindex: len(values) == len(timesteps).

Not sure why we have this difference, suggestions?

Reply:
It is because volumes uses "periods" while the other timeseries uses "timesteps". We should change the timeseries that use periods to do that directly, instead of indirectly as the period between previous timestep and next timestep. In a timeseries with 10 timesteps, there are 9 periods (=which represent volumes). It is easy to do mistakes, misunderstand etc, so reasons to fix that. However, as it is now, that is why I suggested the alternative check above.